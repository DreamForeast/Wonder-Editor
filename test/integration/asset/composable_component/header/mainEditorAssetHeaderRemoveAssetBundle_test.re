open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

open Js.Promise;

let _ =
  describe("MainEditorAssetHeader->remove assetBundle", () => {
    let sandbox = getSandboxDefaultVal();

    beforeEach(() => {
      sandbox := createSandbox();
      MainEditorSceneTool.initState(~sandbox, ());
      MainEditorAssetTool.buildFakeFileReader();
    });

    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe(
      {|
        select assetBundle;
        click remove-button;
            |},
      () =>
      testPromise("should remove it from assetTreeRoot", () =>
        MainEditorAssetUploadTool.loadOneAssetBundle()
        |> then_(uploadedAssetBundleNodeId => {
             MainEditorAssetHeaderOperateNodeTool.removeAssetBundleNode(
               ~assetBundleNodeId=uploadedAssetBundleNodeId,
               (),
             );

             BuildComponentTool.buildAssetChildrenNode()
             |> ReactTestTool.createSnapshotAndMatch
             |> resolve;
           })
      )
    );
  });