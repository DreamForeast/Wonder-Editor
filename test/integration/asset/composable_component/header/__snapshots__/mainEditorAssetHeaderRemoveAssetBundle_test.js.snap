// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorAssetHeader->remove assetBundle 
        select assetBundle;
        click remove-button;
             should remove it from assetTreeRoot 1`] = `
<article
  className="wonder-asset-assetChildren"
/>
`;
